simulation/vis_simulate.pro

```diff
@@ -157,11 +159,21 @@ FUNCTION vis_simulate,obs,status_str,psf,params,jones,skymodel,file_path_fhd=fil
       
     endif ;; end if n_elements(model_image_cube) gt 0 or n_elements(model_uvf_cube) gt 0 or keyword_set(include_eor)
     
+    ;; If bubble file is included, add to the model_uvf_arr
+    IF Keyword_Set(bubble_fname) THEN BEGIN 
+       bubble_uvf = eor_bubble_sim(obs, jones, select_radius=select_radius, bubble_fname=bubble_fname)
+       IF Min(Ptr_valid(bubble_uvf)) GT 0 THEN BEGIN
+         IF n_elements(model_uvf_arr) GT 0 then begin
+           FOR pol_i=0,n_pol-1 DO *model_uvf_arr[pol_i] += *bubble_uvf[pol_i];*uv_mask_use
+         endif else model_uvf_arr = Pointer_copy(bubble_uvf)
+       ENDIF
+    ENDIF 
+
     ;; If diffuse model is included, calculate its contribution and add to the source_model_uv_arr
     IF Keyword_Set(diffuse_model) THEN BEGIN
       IF file_test(diffuse_model) EQ 0 THEN diffuse_model=(file_search(diffuse_model+'*'))[0]
          print,"Reading diffuse model file: "+diffuse_model
-      diffuse_model_uv=fhd_diffuse_model(obs,jones,skymodel,spectral_model_arr=diffuse_spectral_model_uv,/uv_return,/diffuse_units_kelvin,model_filepath=diffuse_model,_Extra=extra)
+      diffuse_model_uv=fhd_diffuse_model(obs,jones,skymodel,spectral_model_arr=diffuse_spectral_model_uv, select_radius=select_radius,/uv_return,/diffuse_units_kelvin,model_filepath=diffuse_model,_Extra=extra)
```

Comment:
If you remove `select_radius` from `fhd_diffuse_model` above as I requested, you will need to remove it here, too.